backend/groth16/bls12-377/verify.go

```diff
@@ -62,21 +61,28 @@ func Verify(proof *Proof, vk *VerifyingKey, publicWitness fr.Vector) error {
 		close(chDone)
 	}()
 
-	if vk.HasCommitment {
-
-		if err := vk.CommitmentKey.Verify(proof.Commitment, proof.CommitmentPok); err != nil {
-			return err
-		}
-
-		publicCommitted := make([]*big.Int, len(vk.PublicCommitted))
-		for i := range publicCommitted {
+	commitmentsSerialized := make([]byte, len(vk.PublicCommitted)*fr.Bytes)
+	for i := range vk.PublicCommitted {
+		publicCommitted := make([]*big.Int, len(vk.PublicCommitted[i]))
```

Comment:
seems the flow is a bit greedy in memory here; each iteration allocate a new slice, convert stuff into big.Int, call `solveCommitmentWire` to convert the big int to `bytes` ... 

Minimizing allocations (including the `commitmentsSerialized` bits) may help; again if we have one or two committed variables we won't see it, but for larger instances.. it will impact perf on the verifier significantly.